compiler/rustc_hir_analysis/src/check/intrinsic.rs

```diff
@@ -314,6 +315,12 @@ pub(crate) fn check_intrinsic_type(
             let type_id = tcx.type_of(tcx.lang_items().type_id().unwrap()).instantiate_identity();
             (0, 0, vec![type_id, type_id], tcx.types.bool)
         }
+        sym::type_of => (
+            0,
+            0,
+            vec![tcx.type_of(tcx.lang_items().type_id().unwrap()).instantiate_identity()],
```

Comment:
```suggestion
            vec![tcx.type_of(tcx.lang_items().type_id().unwrap()).no_bound_vars().unwrap()],
```
so that it ICEs instead if `TypeId` for some reason ever gets generic parameters?